diff --git a/nginx-operator/bundle.Dockerfile b/nginx-operator/bundle.Dockerfile
index 132db509b..3faedffc2 100644
--- a/nginx-operator/bundle.Dockerfile
+++ b/nginx-operator/bundle.Dockerfile
@@ -1,3 +1,15 @@
+ARG BUILDER_IMG=alpine:latest
+FROM ${BUILDER_IMG} AS builder
+
+# Copy files to locations specified by labels.
+COPY bundle/manifests /manifests/
+COPY bundle/metadata /metadata/
+COPY bundle/tests/scorecard /tests/scorecard/
+
+# Replace MK8S_VERSION_STUB with the actual version.
+ARG METALK8S_VERSION=MK8S_VERSION_STUB
+RUN sed -i "s/MK8S_VERSION_STUB/${METALK8S_VERSION}/g" /manifests/nginx-operator.clusterserviceversion.yaml
+
 FROM scratch
 
 # Core bundle labels.
@@ -14,7 +26,7 @@ LABEL operators.operatorframework.io.metrics.project_layout=helm.sdk.operatorfra
 LABEL operators.operatorframework.io.test.mediatype.v1=scorecard+v1
 LABEL operators.operatorframework.io.test.config.v1=tests/scorecard/
 
-# Copy files to locations specified by labels.
-COPY bundle/manifests /manifests/
-COPY bundle/metadata /metadata/
-COPY bundle/tests/scorecard /tests/scorecard/
+# Copy files from builder to locations specified by labels.
+COPY --from=builder /manifests /manifests/
+COPY --from=builder /metadata /metadata/
+COPY --from=builder /tests/scorecard /tests/scorecard/
